api/v1beta2/network_types.go

```diff
@@ -94,12 +104,57 @@ var (
 
 // TargetGroupHealthCheck defines health check settings for the target group.
 type TargetGroupHealthCheck struct {
-	Protocol        *string `json:"protocol,omitempty"`
-	Path            *string `json:"path,omitempty"`
-	Port            *string `json:"port,omitempty"`
-	IntervalSeconds *int64  `json:"intervalSeconds,omitempty"`
-	TimeoutSeconds  *int64  `json:"timeoutSeconds,omitempty"`
-	ThresholdCount  *int64  `json:"thresholdCount,omitempty"`
+	// The protocol to use to health check connect with the target.
+	// Defaults to TCP.
+	// +kubebuilder:default=TCP
+	// +kubebuilder:validation:Enum=TCP;HTTP;HTTPS
+	// +optional
+	Protocol *string `json:"protocol,omitempty"`
+
+	// The destination for health checks on the targets when using the protocol HTTP or HTTPS,
+	// otherwise the path will be ignored.
+	// Defaults to "/readyz".
+	// +kubebuilder:default="/readyz"
```

Comment:
Why readyz vs healthz? This also poses the question if we should have the default to be kubernetes specific

Reply:
healthz is deprecated https://kubernetes.io/docs/reference/using-api/health-checks/#api-endpoints-for-health

They should be kubernetes specific because this is a check against the kube-apiserver component